### DISCUSSION TITLE
A few minor improvements to some disk images

### DISCUSSION THREAD

**Discussion by kiwisincebirth** [wwarthen/RomWBW#469] [Ideas]
@wwarthen - 

A few minor improvements to some disk images

* ZP.COM - currently only in ZPM3 disk image - move the utility to common/Z3 as works for any ZCPR3 based OS
* Z80ASM (Disk Image) - Only includes Z80ASM assembler. Should include the full SLR systems toolset, the SLRNK (linker), companion 180ASM assembler, and the disassembler from same author
* COMBO (Disk Image) - Add Z3PLUS (full os), either at end as a new slice -or- replace wordstar with Z3PLUS to keep combo image size the same. Main reason ensure the combo imgae has the complete set of OS images

Happy to action these if in agreement.

Mark.

**Comment by wwarthen** [comment]
Hi @kiwisincebirth,

> A few minor improvements to some disk images
> 
> * ZP.COM - currently only in ZPM3 disk image - move the utility to common/Z3 as works for any ZCPR3 based OS

I don't know where ZP.COM came from.  I think it is duplicative of ZAP.COM which runs on all OSes and seems to be at least as capable.  I would be inclined to just remove ZP.COM.  If you want to keep it and move it to Common/Z3, that is fine with me.

> * Z80ASM (Disk Image) - Only includes Z80ASM assembler. Should include the full SLR systems toolset, the SLRNK (linker), companion 180ASM assembler, and the disassembler from same author

Definitely.

> * COMBO (Disk Image) - Add Z3PLUS (full os), either at end as a new slice -or- replace wordstar with Z3PLUS to keep combo image size the same. Main reason ensure the combo imgae has the complete set of OS images

There are actually 2 OSes that are not in the Combo image.  In addition to Z3PLUS, there is also QDOS.  I don't want to change the slice layout of the combo image at this point -- too many people expect the current layout.  The layout may need to change someday, but that would be a very big deal and I would need to be convinced that the new layout would not change for a very long time.

So, I guess the additional OSes could be added to the end of the existing combo image slices.  However, this has issues as well:

- Spencer distributes CF Cards that are 64MB.  Adding slices to the combo image will cause it to not fit.  This is not a real problem because any slices exceeding the end of the media will just be unusable.  However, I believe I will get a huge number of support queries when the tools that copy images to media issue ugly warnings/errors.  Unfortunately, people don't read the documentation, especially if they have already done this before and expect it to work the same.

- Some people have started using slices after the 6 slices of the combo image for their own data.  They have the expectation that appying a new combo image will only overlay the first 6 slices.  I have occasionally suggested that people start their own slices at 32 to avoid this in the future, but many have not done this.  Applying a larger combo image will destroy their existing data if they are not careful.

I completely understand that these two issues can easily be worked around by reading documentation and/or use of COPYSL.  The problem is not reading documentation.

> Happy to action these if in agreement.

Would be great if you want to address the first two items.  The third item requires more discussion.

Thanks, Wayne

**Comment by kiwisincebirth** [comment]
@wwarthen - ZP.COM - Is quite seperate to superzap - It is a lot more sophisticated, HLP and CFG, memory editor, uses Z3 terminal capabilities, line drawing - I am inclined to keep/move it.

```09:42 A15:ROOT>zp /?
ZP v1.7a ZCPR33+ Patcher (Type 3 at 8000h)
Syntax:
   Memory  ZP [hex_addr]
   File    ZP [dir:]afn
   Disk    ZP dir:
```

> ZP17a - ZCPR3 screen-oriented file/disk/memory record patcher uses the ZPATCH command set and requires a VLIB4D+ extended Z3TCAP. One- record cache can be exchanged with a file, disk, or memory record. 'HP' RPN integer calculator may be called up at any time for hex, decimal, binary or character mode calculations. Version 17a fixes some cache mode bugs, an edit mode screen refresh bug and corrects the return references in the HP help files. See ZP17.HZP.

Found Here
https://oldcomputers.dyndns.org/public/pub/cdrom/walnut_creek_cdrom/jsage/znode3/uploads/index.html

**Comment by wwarthen** [comment]
OK on `ZP.COM`.  I didn't look at it that deeply -- initially looked like `ZAP`.  Fine to keep it.

Thanks, Wayne

**Comment by kiwisincebirth** [comment]
See PR #471 and #472

**Comment by kiwisincebirth** [comment]
Closing this. Combo Image changes extracted to a new discussion
